### ISSUE TITLE
learning rate setting when training 12net-cls-only.caffemodel

### ISSUE THREAD

**Issue by bpf476499866** [CongWeilin/mtcnn-caffe#41]
Sorry I met another problem...errrrrr....QAQ

when I was training 【12-net-cls-only.caffemodel】without fine-tune, 
(to manage this, I change <12net/pythonLayer.py, line 38> into 【loss_task = random.randint(0**,0**)】, is it right?)

No matter which learning policy I choose in <solver>,  the loss curve result in so ugly an appearence.
Are these loss curves below normal? ( because the p-net is truly a tiny net ) 

Or maybe this caused by some mistakes nothing to do with <solver> but with some other operating?
Could you please share some advice for training 12net-cls-only.caffemodel。。。

![a](https://cloud.githubusercontent.com/assets/16606917/26520723/9781e2c8-430a-11e7-84cf-3f9ad6bec51f.png)
![b](https://cloud.githubusercontent.com/assets/16606917/26520735/c97fd0e6-430a-11e7-82b3-113fb147c612.png)
![c](https://cloud.githubusercontent.com/assets/16606917/26520738/d21f52f8-430a-11e7-8d04-cfb3ddd2ffdd.png)
![d](https://cloud.githubusercontent.com/assets/16606917/26520740/d638f16e-430a-11e7-9963-afe2f786b94d.png)
![e](https://cloud.githubusercontent.com/assets/16606917/26520762/442a139c-430b-11e7-80cd-4b217fa8f484.png)
![f](https://cloud.githubusercontent.com/assets/16606917/26520764/49b0f1d2-430b-11e7-8f92-4a8da09aa70e.png)
![g](https://cloud.githubusercontent.com/assets/16606917/26520767/503be0b6-430b-11e7-8fd6-fefa7e81b4c8.png)

**Comment by bpf476499866** [comment]
@CongWeilin 
Besides, do I need to add an "ignored lable = -1" parameter in cls_loss layer ( type: "SoftmaxWithLoss) in 12-net-train.prototxt, so as to let cls_loss layer ignores the imput when roi.imdb's data is sent into net? 
Becase when fine-tuning roi.imdb with cls-only.caffemodel, the loss switch between very high and very low as task switch between cls and roi.
When roi.imdb send into net, both class-loss and regression loss will exist, the roi loss is averaging 0.04, yet the cls loss goes to 0.7 comparing with cls.imdb's class-loss wich is around 0.1~0.2 (as images above)

**Comment by sizhangyu** [comment]
@bpf476499866 Have you got any progress? My training loss does not decrease to a small value.

**Comment by bpf476499866** [comment]
@sizhangyu 
sadly no : ((((((
My implement still staying in the green loss feature above…… the trained roi-task could not capture face rect at all

**Comment by wolf345** [comment]
@bpf476499866 @CongWeilin @sizhangyu  cls_loss = 0.693147?why?

**Comment by bpf476499866** [comment]
I don't know  X: (   
did you face the same problem?
